mock-pop-mining/src/main/java/org/veriblock/sdk/mock/VeriBlockBlockchain.java

```
@@ -15,37 +15,98 @@
+                                                            secondPreviousKeystone,
+                                                            blockData.getMerkleRoot(),
+                                                            Utils.getCurrentTimestamp(),
+                                                            // FIXME: use the difficulty calculator to set the correct difficulty
+                                                            chainHead.getDifficulty(),
+                                                            nonce);
+
+                ValidationService.checkProofOfWork(newBlock);
+
+                add(newBlock);
+                blockDataStore.put(newBlock.getHash(), blockData);
+
+                return newBlock;
 
-        VeriBlockBlock newBlock = new VeriBlockBlock(chainHead.getHeight() + 1,
-                                                     chainHead.getVersion(),
-                                                     chainHead.getHash().trimToPreviousBlockSize(),
-                                                     // FIXME: set the keystones correctly
-                                                     VBlakeHash.EMPTY_HASH.trimToPreviousKeystoneSize(),
-                                                     VBlakeHash.EMPTY_HASH.trimToPreviousKeystoneSize(),
-                                                     blockData.getMerkleRoot(),
-                                                     chainHead.getTimestamp() + 1,
-                                                     // FIXME: use the difficulty calculator to set the correct difficulty
-                                                     chainHead.getDifficulty(),
-                                                     0);
-        add(newBlock);
-        blockDataStore.put(newBlock.getHash(), blockData);
-
-        return newBlock;
+            } catch (VerificationException e) {
+            }
```

do not ignore exceptions, at least log them

---

this exception is how SerializeService signals that the block validation has failed(and thus we need to try another nonce)